src/components/v2/MintRepayVai/Mint/index.tsx

```diff
@@ -0,0 +1,127 @@
+/** @jsxImportSource @emotion/react */
+import React from 'react';
+import BigNumber from 'bignumber.js';
+import Typography from '@mui/material/Typography';
+
+import { convertWeiToCoins } from 'utilities/common';
+import { Icon } from '../../Icon';
+import { TokenTextField } from '../../TokenTextField';
+import { SecondaryButton } from '../../Button';
+import { useStyles } from './styles';
+
+const VAI_SYMBOL = 'vai';
+
+export interface IMintUiProps {
+  disabled: boolean;
+  isSubmitting: boolean;
+  onSubmit: (value: BigNumber) => Promise<void>;
+  limitWei: BigNumber;
+  mintFeePercentage: number;
+}
+
+// TODO: Move to dashboard component/container once created
+export const MintUi: React.FC<IMintUiProps> = ({
+  disabled,
+  limitWei,
+  mintFeePercentage,
+  isSubmitting,
+  onSubmit,
+}) => {
+  const styles = useStyles();
+  const [value, setValue] = React.useState<BigNumber | ''>('');
+
+  const isValueValid = value && value.isGreaterThan(0);
+
+  // Convert limit into VAI
+  const readableLimitVai = React.useMemo(
+    () => convertWeiToCoins({ value: limitWei, tokenSymbol: VAI_SYMBOL }).toString(),
+    [limitWei],
+  );
+
+  // Calculate fee
+  const feeWei = new BigNumber(value || 0).multipliedBy(mintFeePercentage).dividedBy(100);
+  const readableFeeVai = convertWeiToCoins({ value: feeWei, tokenSymbol: VAI_SYMBOL }).toString();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    // Prevent native submission behavior
+    e.preventDefault();
+
+    if (value) {
+      await onSubmit(value);
+
+      // Reset value
+      setValue('');
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <TokenTextField
+        css={styles.textField}
+        tokenSymbol={VAI_SYMBOL}
+        value={value}
+        onChange={setValue}
+        maxWei={limitWei}
+        disabled={disabled || isSubmitting}
+        rightMaxButtonLabel="SAFE MAX"
+      />
+
+      <div css={styles.getRow({ isLast: false })}>
+        <div css={styles.infoColumn}>
+          <Icon name={VAI_SYMBOL} css={styles.coinIcon} />
+          <Typography component="span" variant="small2">
+            Available VAI Limit
+          </Typography>
+        </div>
+
+        <Typography component="span" css={styles.infoValue} variant="small1">
+          {readableLimitVai} VAI
+        </Typography>
+      </div>
+
+      <div css={styles.getRow({ isLast: true })}>
```

Comment:
nit: Prefixing style functions is a new convention. I could see us not having any preference about this or preferring to use or not use it.

Reply:
Yeah, I've been doing that to differentiate functions from interpolated css strings. To me, it doesn't make a huge difference whether we all follow this convention or not (I think it's pretty clear to devs what this does either way), but if you want to follow another pattern I'm happy to oblige.
Just let me know what you prefer.